action.yaml

```diff
@@ -5,19 +5,28 @@ inputs:
     description: Options to pass to `prek run`
     required: false
     default: '--all-files'
+  prek-version:
+    description: Version of prek to install (e.g., '0.2.1', 'latest')
+    required: false
+    default: 'latest'
 runs:
   using: composite
   steps:
   - name: Install prek
-    run: python -m pip install prek
+    run: |
+      if [ "${{ inputs.prek-version }}" = "latest" ]; then
+        python -m pip install prek
+      else
+        python -m pip install prek==${{ inputs.prek-version }}
+      fi
     env:
       PIP_DISABLE_PIP_VERSION_CHECK: 1
       PIP_PROGRESS_BAR: 'off'
     shell: bash
   - uses: actions/cache@0400d5f644dc74513175e3cd8d07132dd4860809 # v4.2.4
     with:
       path: ~/.cache/prek
-      key: prek-v1|${{ runner.os }}|${{ runner.arch }}|${{ hashFiles('**/.pre-commit-config.yaml') }}
+      key: prek-v1|${{ inputs.prek-version }}|${{ runner.os }}|${{ runner.arch }}|${{ hashFiles('**/.pre-commit-config.yaml') }}
```

Comment:
Analyze the good and bad about adding prek version to cache key? What if the cache layout is compatible in most prek releases? Does it necessary to expose the internal cache version here for the cache key?